### ISSUE TITLE
Locating source file for findbugs violation

### ISSUE THREAD

**Issue by vara-prasad** [SonarQubeCommunity/sonar-findbugs#47]
In sonar-findbugs-plugin-3.4.3, source file is always searched relative to "src/main/java/"
In latest master branch, this check is enhanced to consider "src/main/java","src/main/webapp","src/main/resources","src"

But there could be cases where we execute sonar scan on projects which will have multiple modules and multiple languages.

Consider the below example for project structure

> project
> project/module1
> project/module1/app1
> project/module1/app2
> project/module2/app3
> project/module2/app4
> project/module3/app5
> project/module3/app6
> project/module4/python-script

In the above module 1 to module 3 will can have java apps and module 4 can have python scripts
There are cases where we scan project with project base directory as "project" and project source directory as "."
In this case, the findbugs violation created will not be translated to sonar violation due to the above restriction of source paths.
Can the search for source file be relaxed to do pattern search (matchesPathPattern) as **/path_to_java_file ?

**Comment by h3xstream** [comment]
```java
Iterable<InputFile> files = fs.inputFiles(fs.predicates().hasRelativePath(sourceDir+"/"+fileName));
```
The predefined list or source directories are **relative path** to the module directory. You should not get issues if your project is multi-modules.

I will check if the wildcard can be used in this context. This would avoid having to hard-coded source directories.

**Comment by vara-prasad** [comment]
@h3xstream in our project configuration the source directory starts from project/. as we give sonar.sources as "." and sonar base directory as "project" and hence may need to use "matchesPathPattern"

**Comment by user82461** [comment]
I ran into the same issue. Our project is not multi-module but sources are in "/src/java", not "/src/main/java". It works if we make a symlink for that path, but that breaks the SCM functionality since the paths no longer match.

Could you add "/src/java" to the array our use the sonar.sources value from the properties file?

**Comment by h3xstream** [comment]
This should have been fixed already by #68
